PowerShell/JumpCloud Module/Public/Users/Remove-JCUser.ps1

```diff
@@ -67,52 +92,106 @@ UserID has an Alias of _id. This means you can leverage the PowerShell pipeline
             } else {
                 Throw "Username does not exist. Run 'Get-JCUser | Select-Object username' to see a list of all your JumpCloud users."
             }
+        } elseif ($PSCmdlet.ParameterSetName -eq 'UserID') {
+            # Validate if the Id is a JC User from the $UserHash
+            if ($UserHash.ContainsKey($UserID)) {
+                Write-Debug "UserID is a valid JumpCloud User"
+                $UserId = $UserHash.GetEnumerator().Where({ $_.Name -contains ($UserID) }).Name
+                $Username = $UserHash.GetEnumerator().Where({ $_.Name -contains ($UserID) }) | Select-Object -ExpandProperty Value | Select-Object -ExpandProperty username
+            } else {
+                Write-Error "User does not exist. Please enter a valid UserID."
+                # Exit the script
+                Exit
+            }
         }
-
-        if ($PSCmdlet.ParameterSetName -eq 'UserID' ) {
-            $Username = $UserID
+        # Check if the user is a manager
+        if ($UserHash.Values.manager -contains ($UserID)) {
+            $isManager = $true
+            # Count the number of users the manager is managing
+            # $managerCount = ($UserHash.Values.manager -eq $UserID).Count
+            # Save each user the manager is managing in a list
+            $managedUsers = $UserHash.GetEnumerator().Where({ $_.Value.manager -eq $UserID }).Name
+            Write-Debug "Manager is managing $managedUsers users"
+            $hasManagerId = Get-JcSdkUser -Id $UserID | Select-Object -ExpandProperty manager
+            Write-Debug "Manager is managed by $hasManagerId"
+        } else {
+            $isManager = $false
+            Write-Debug "User is not a manager"
         }
 
         if (!$force) {
-            try {
-                $URI = "$JCUrlBasePath/api/systemusers/$UserID"
-                Write-Warning "Are you sure you wish to delete user: $Username ?" -WarningAction Inquire
-                $delete = Invoke-RestMethod -Method Delete -Uri $URI -Headers $hdrs -UserAgent:(Get-JCUserAgent)
-                $Status = 'Deleted'
-            } catch {
-                $Status = $_.ErrorDetails
-            }
-
-            $FormattedResults = [PSCustomObject]@{
-                'User'    = $Username
-                'Results' = $Status
+            if ($CascadeManager -and $isManager) {
```

Comment:
Let's use $PSBoundParameters['CascadeManager'] to keep things consistent and avoid accidental variable assignment triggering this code section